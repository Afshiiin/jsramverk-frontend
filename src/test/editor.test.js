import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Editor from '../components/body/editor/Editor'; // Adjust the import path as needed

test('Check the section where the list of existing files loads and verify the header of the div', async () => {
  render(<Editor />);
  
  const editorElement = await screen.findByTestId('ListOfFiles');
  expect(editorElement).toBeInTheDocument();
  expect(editorElement).toHaveTextContent('Existing files');
});


jest.mock('@ckeditor/ckeditor5-react', () => ({
  CKEditor: ({ onChange }) => (
    <textarea
      data-testid="mockedCKEditor"
      onChange={(e) => onChange({}, { getData: () => e.target.value })}
    />
  ),
}));

test('updates CKEditor value on change', () => {
  render(<Editor />);

  const editor = screen.getByTestId('mockedCKEditor');
  fireEvent.change(editor, { target: { value: 'My name is Afshin' } });

  expect(editor.value).toBe('My name is Afshin');
});



/* test('shows alert message when trying to create a file with an empty name', async () => {
  render(<Editor />);

  const input = screen.getByTestId('inputTitleDoc').querySelector('input');

  fireEvent.change(input, { target: { value: '' } });

  const createButton = screen.getByText(/Create New File/i);
  fireEvent.click(createButton);

  const alertMessage = await screen.findByText(/Give a name to the file!/i);
  expect(alertMessage).toBeInTheDocument();
}); */




 
