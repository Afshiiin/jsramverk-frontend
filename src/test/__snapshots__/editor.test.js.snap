// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter function should show items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiPaper-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused Mui-required Mui-required"
                data-shrink="true"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Enter file name
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="outlined-basic"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              style="margin-top: 1.1%; text-transform: none; border-left: 1px solid black;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Create New File 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="NoteAddIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm2 14h-3v3h-2v-3H8v-2h3v-3h2v3h3zm-3-7V3.5L18.5 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              style="margin-top: 1.1%; text-transform: none; border-left: 1px solid black;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Show In Console 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="FactCheckIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2M10 17H5v-2h5zm0-4H5v-2h5zm0-4H5V7h5zm4.82 6L12 12.16l1.41-1.41 1.41 1.42L17.99 9l1.42 1.42z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div>
          <div
            style="width: 80%; border: 1px solid black; min-height: 81vh; display: inline-block;"
          >
            <div />
          </div>
          <div
            style="width: 19.47%; display: inline-block; position: absolute; min-height: 81vh;"
          >
            <div
              style="border-bottom: 1px solid black; text-align: center;"
            >
              <h3>
                Existing files
              </h3>
            </div>
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
              style="max-height: 54.8vh; overflow: auto; text-align: center;"
            >
              <ul
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                There is no saved file to show
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiPaper-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Enter file name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="outlined-basic"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            style="margin-top: 1.1%; text-transform: none; border-left: 1px solid black;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create New File 
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="NoteAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm2 14h-3v3h-2v-3H8v-2h3v-3h2v3h3zm-3-7V3.5L18.5 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            style="margin-top: 1.1%; text-transform: none; border-left: 1px solid black;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Show In Console 
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="FactCheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2M10 17H5v-2h5zm0-4H5v-2h5zm0-4H5V7h5zm4.82 6L12 12.16l1.41-1.41 1.41 1.42L17.99 9l1.42 1.42z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div>
        <div
          style="width: 80%; border: 1px solid black; min-height: 81vh; display: inline-block;"
        >
          <div />
        </div>
        <div
          style="width: 19.47%; display: inline-block; position: absolute; min-height: 81vh;"
        >
          <div
            style="border-bottom: 1px solid black; text-align: center;"
          >
            <h3>
              Existing files
            </h3>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            style="max-height: 54.8vh; overflow: auto; text-align: center;"
          >
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            >
              There is no saved file to show
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
